File: java-api-injection/src/main/java/Java_Api_Injection/controller/ImageController.java
package Java_Api_Injection.controller;

import org.springframework.web.bind.annotation.*;

import Java_Api_Injection.model.Image;
import Java_Api_Injection.service.ImageService;

import java.util.List;

@RestController()
@RequestMapping("image")
public class ImageController {
  ImageService imageService = new ImageService();

  @GetMapping()
  public List<Image> getAll() {
    return imageService.getAll();
  }

  @GetMapping("user/{userId}")
  public List<Image> getAllByUser(@PathVariable int userId) {

    System.out.println(userId);
    return imageService.getAllByUser(userId);
  }

  @GetMapping("{id}")
  public Image get(@PathVariable int id) {
    return imageService.get(id);
  }

  @GetMapping("search/{userId}")
  public List<Image> search(@PathVariable int userId, @RequestParam String input) {
    System.out.println(input);

    return imageService.search(userId, input);
  }
}

File: node-api-injection/api/src/controllers/ImageController.js
const express = require('express')
const router = express.Router()
const imageService = require('../services/ImageService.js')
const route = 'image'

router.get(`/${route}`, async (req, res) => {
  return res.status(200).json(await imageService.getAll())
})

router.get(`/${route}/user/:userId`, async (req, res) => {
  const { userId } = req.params

  return res.status(200).json(await imageService.getAllByUser(userId))
})

router.get(`/${route}/:id`, async (req, res) => {
  const { id } = req.params

  return res.status(200).json(await imageService.get(id))
})

router.get(`/${route}/search/:userId`, async (req, res) => {
  const { userId } = req.params
  const { input } = req.query

  return res.status(200).json(await imageService.search(userId, input))
})

module.exports = router
